packages/core/redux/mainWallet/actions.js

```diff
@@ -141,27 +134,46 @@ const handleToken = (token: TokenModel) => async (dispatch, getState) => {
       }
     })
     .on(EVENT_UPDATE_BALANCE, ({ account, balance }) => {
-      const wallets = getState().get(DUCK_ETH_MULTISIG_WALLET)
-      if (wallets.item(account)) {
-        dispatch({
-          type: ETH_MULTISIG_BALANCE,
-          walletId: account,
-          balance: new BalanceModel({
-            id: token.id(),
-            amount: new Amount(balance, token.symbol(), true),
-          }),
-        })
-      } else {
-        const addresses = getMainEthWallet(getState())
-        if (addresses.includes(account)) {
-          dispatch({
-            type: WALLET_TOKEN_BALANCE,
-            balance: new BalanceModel({
-              id: token.id(),
-              amount: new Amount(balance, token.symbol()),
-            }),
-          })
-        }
+
+      switch (token.blockchain()) {
+        case BLOCKCHAIN_ETHEREUM:
+          const wallets = getState().get(DUCK_ETH_MULTISIG_WALLET)
+          if (wallets.item(account)) {
+            dispatch({
+              type: ETH_MULTISIG_BALANCE,
+              walletId: account,
+              balance: new BalanceModel({
+                id: token.id(),
+                amount: new Amount(balance, token.symbol(), true),
+              }),
+            })
+          } else {
+            const addresses = getMainEthWallet(getState())
+            if (addresses.includes(account)) {
+              dispatch({
+                type: WALLET_TOKEN_BALANCE,
+                balance: new BalanceModel({
+                  id: token.id(),
+                  amount: new Amount(balance, token.symbol()),
+                }),
+              })
+            }
+          }
+          break
+
+        case BLOCKCHAIN_NEM:
+        case BLOCKCHAIN_BITCOIN:
+        case BLOCKCHAIN_BITCOIN_CASH:
+        case BLOCKCHAIN_LITECOIN:
+        case BLOCKCHAIN_WAVES:
+          const wallet = getWallet(token.blockchain(), account)(getState())
+          dispatch({ type: WALLETS_UPDATE_BALANCE, walletId: wallet.id, balance: new Amount(balance, token.symbol()) })
+          break
+
+        default:
+          //eslint-disable-next-line
```

Comment:
Please do not disable all ESLint rules.
If you need `console.log`, just disable it: `// eslint-disable-next-line no-console`